src/Util/ClassSourceManipulator.php

```diff
@@ -0,0 +1,992 @@
+                    ->getNode()
+                ;
+            case self::CONTEXT_CLASS_METHOD:
+                return new Node\Expr\Variable('__EXTRA__LINE');
+            default:
+                throw new \Exception('Unknown context: '.$context);
+        }
+    }
+
+    private function createSingleLineCommentNode(string $comment, string $context)
+    {
+        $this->pendingComments[] = $comment;
+        switch ($context) {
+            case self::CONTEXT_OUTSIDE_CLASS:
+                // just not needed yet
+                throw new \Exception('not supported');
+            case self::CONTEXT_CLASS:
+                // just not needed yet
+                throw new \Exception('not supported');
+            case self::CONTEXT_CLASS_METHOD:
+                return new Node\Expr\Variable('__COMMENT__VAR_'.(count($this->pendingComments) - 1));
+            default:
+                throw new \Exception('Unknown context: '.$context);
+        }
+    }
+
+    private function createDocBlock(array $commentLines)
+    {
+        $docBlock = "/**\n";
+        foreach ($commentLines as $commentLine) {
+            $docBlock .= " * $commentLine\n";
+        }
+        $docBlock .= "\n */";
+
+        return $docBlock;
+    }
+
+    private function addMethod(Node\Stmt\ClassMethod $methodNode)
+    {
+        $classNode = $this->getClassNode();
+        $methodName = $methodNode->name;
+        $existingIndex = null;
+        if ($this->methodExists($methodName)) {
+            if (!$this->overwrite) {
+                $this->writeNote(sprintf(
+                    'Not generating <info>%s::%s()</info>: method already exists',
+                    Str::getShortClassName($this->getThisFullClassName()),
+                    $methodName
+                ));
+
+                return;
+            }
+
+            // record, so we can overwrite in the same place
+            $existingIndex = $this->getMethodIndex($methodName);
+        }
+
+
+        $newStatements = [];
+
+        // put new method always at the bottom
+        if (!empty($classNode->stmts)) {
+            $newStatements[] = $this->createBlankLineNode(self::CONTEXT_CLASS);
+        }
+
+        $newStatements[] = $methodNode;
+
+        if (null === $existingIndex) {
+            // just them on the end!
+
+            $classNode->stmts = array_merge($classNode->stmts, $newStatements);
+        } else {
+            array_splice(
+                $classNode->stmts,
+                $existingIndex,
+                1,
+                $newStatements
+            );
+        }
+
+        $this->updateSourceCodeFromNewStmts();
+    }
+
+    private function getEntityTypeHint($doctrineType)
+    {
+        switch ($doctrineType) {
+            case 'string':
+            case 'text':
+            case 'guid':
+                return 'string';
+
+            case 'array':
+            case 'simple_array':
+            case 'json_array':
+            case 'json':
```

Comment:
not good. You can store other things than arrays in a `json` field (even though it is the common need)

Reply:
Actually, check out the JsonType class - it uses `$val = json_decode($value, true);` in `convertToPHPValue`. So it looks like it *does* always return an array.

However, that method could also return `null`. So probably having no return type is safer.

Reply:
nope, it can also return any scalar or null. The `true` argument just means it will never return you a `stdClass`